File: src/main/java/tomeko/legacyskyblock/dungeons/HideDamageSplash.java
package tomeko.legacyskyblock.dungeons;

import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents;
import net.minecraft.client.Minecraft;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.decoration.ArmorStand;
import org.polyfrost.oneconfig.api.config.v1.annotations.Switch;
import tomeko.legacyskyblock.config.LegacySkyblockConfig;
import tomeko.legacyskyblock.utils.HypixelPackets;

public class HideDamageSplash {
    private static final String SUBCATEGORY_HIDE_DAMAGE_SPLASH = "Hide Damage Splash";

    @Switch(
            title = "Enabled",
            category = LegacySkyblockConfig.CATEGORY_DUNGEONS,
            subcategory = SUBCATEGORY_HIDE_DAMAGE_SPLASH
    )
    public boolean enabled = false;

    @Switch(
            title = "Work Outside Dungeons",
            category = LegacySkyblockConfig.CATEGORY_DUNGEONS,
            subcategory = SUBCATEGORY_HIDE_DAMAGE_SPLASH
    )
    public boolean workOutsideDungeons = false;

    public HideDamageSplash() {
        ClientTickEvents.END_CLIENT_TICK.register(this::hideSplash);
    }

    private void hideSplash(Minecraft client) {
        if (!enabled
                || !HypixelPackets.inSkyblock
                || (!HypixelPackets.inDungeons && !workOutsideDungeons)
                || client.isPaused()
                || client.level == null
        ) return;

        Iterable<Entity> entities = client.level.entitiesForRendering();
        for (Entity entity : entities) {
            if (entity == null) continue;

            if (entity instanceof ArmorStand armorStandEntity) {
                String name = armorStandEntity.getName().getString().replace(",", "").replace(".", "");

                if (name.endsWith("k") || name.endsWith("M")) {
                    name = name.substring(0, name.length() - 1);
                }

                try {
                    Long.parseLong(name);
                    entity.discard();
                    continue;
                } catch (NumberFormatException ignored) {
                }

                if (name.length() >= 2) {
                    name = name.substring(1, name.length() - 1);
                }

                if (name.endsWith("k") || name.endsWith("M")) {
                    name = name.substring(0, name.length() - 1);
                }

                try {
                    Long.parseLong(name);
                    entity.discard();
                } catch (NumberFormatException ignored) {
                }
            }
        }
    }
}

File: src/main/kotlin/tomeko/legacyskyblock/dungeons/HideDamageSplash.kt
package tomeko.legacyskyblock.dungeons

import net.fabricmc.fabric.api.client.event.lifecycle.v1.ClientTickEvents
import net.minecraft.client.Minecraft
import net.minecraft.world.entity.Entity
import net.minecraft.world.entity.decoration.ArmorStand
import tomeko.legacyskyblock.config.LegacySkyblockConfig
import tomeko.legacyskyblock.utils.HypixelPackets

object HideDamageSplash {
    fun register() {
        ClientTickEvents.END_CLIENT_TICK.register(ClientTickEvents.EndTick { mc: Minecraft -> hideSplash(mc) })
    }

    private fun hideSplash(mc: Minecraft) {
        if (
            !HypixelPackets.inSkyblock
            || mc.isPaused
            || mc.level == null
            || HypixelPackets.currentIsland == null
            || !LegacySkyblockConfig.hideDamageSplashEnabledIslands[HypixelPackets.currentIsland!!.ordinal]
        ) return


        val entities: Iterable<Entity?> = mc.level!!.entitiesForRendering()
        for (entity in entities) {
            if (entity == null) continue

            if (entity is ArmorStand) {
                var name = entity.name.string.replace(",", "").replace(".", "")

                if (name.endsWith("k") || name.endsWith("M")) {
                    name = name.substring(0, name.length - 1)
                }

                try {
                    name.toLong()
                    entity.discard()
                    continue
                } catch (_: NumberFormatException) {
                }

                if (name.length >= 2) {
                    name = name.substring(1, name.length - 1)
                }

                if (name.endsWith("k") || name.endsWith("M")) {
                    name = name.substring(0, name.length - 1)
                }

                try {
                    name.toLong()
                    entity.discard()
                } catch (_: NumberFormatException) {
                }
            }
        }
    }
}
